cmd/jaeger/internal/extension/jaegerquery/server_test.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2024 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jaegerquery
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/component"
+	"go.uber.org/zap"
+)
+
+// mockHost implements the Host interface for testing purposes
+type mockHost struct {
+	logger zap.Logger
+	// Captures arguments for expected methods
+	reportFatalErrorCalled     bool
+	reportFatalErrorArg        error
+	getFactoryCalled           bool
+	getFactoryKindArg          component.Kind
+	getFactoryComponentTypeArg component.Type
+	getExtensionsCalled        bool
```

Comment:
I don't see you using any of these. If you don't need special behavior of the host you can use Nop available in OTEL (you can check other tests in parallel packages using it)

Reply:
Removed the unnecessary variables in the latest version